kolibri/core/content/api.py

```diff
@@ -862,6 +863,14 @@ class ContentNodeViewset(InternalContentNodeMixin, RemoteMixin, ReadOnlyValuesVi
     def retrieve(self, request, pk=None):
         if pk is None:
             raise Http404
+
+        try:
```

Comment:
It seems like I linked the wrong viewset to implement this in the issue. We would want to implement this in ContentNodeTreeViewset not in the ContentNodeViewset and hence the failing tests. Could you make the changes in ContentNodeTreeViewset instead? That should fix the tests.

Reply:
Ok 👍

Reply:
@ozer550 it does not give me 404 when PK is None after moving the try block to ContentNodeTreeViewset 
Currently I am using 400 response when pk is None

I need a clarification from your side.

Reply:
For which viewset are we talking about here?

Reply:
ContentNodeViewset

Reply:
Hmm.. Thats wierd cause we clearly have a check for this:
```
if pk is None:
    raise Http404
```

Reply:
yeah I am also confused at that point

Reply:
ohh I see the test `test_contentnode_tree_none_pk` is checking for 400 we should check 404 as we have an explicit if block mentioned above in ContentNodeTreeViewset testing that. But in the other one None case would be caught first so we would check for 400 there.

Reply:
Could you change the test to this:

```
    def test_contentnode_tree_none_pk(self):
        response = self.client.get("/api/content/contentnode_tree/")
        self.assertEqual(response.status_code, 404)
        
  ```

Reply:
Ok I will do it 👍

Reply:
The reason for the confusion is how the reverse function works. When you pass None for the pk parameter in the reverse() function, it generates a URL with None literally inserted, or it may fail to match a valid URL pattern. This issue arises before the view logic is even reached. Django interprets the malformed URL or missing parameter as a Bad Request 400 error, which is why test receives a 400 response.